drivers/hmis/app/models/hmis/hud/enrollment.rb

```diff
@@ -281,6 +281,65 @@ def self.generate_enrollment_id
     generate_uuid
   end
 
+  # Data Collection Features that are enabled for this enrollment, thru its project (e.g. Current Living Situation)
+  #
+  # Is it enabled?
+  #   If ANY data exists for it in this enrollment, even if no active instances exist, then yes.
+  #   (This supports both inactive instances and migrated-in data, as well as context changes such as change in HoH)
+  #
+  # Who is data collected about?
+  #   Choose the "best" instance – IE the one that would actually be selected
+  #   when creating a new record – and return the data_collected_about on it.
+  #
+  # Returns an OpenStruct that is resolved by the DataCollectionFeature GQL type.
+  def data_collection_features
+    # Create OpenStruct for each enabled feature
+    Hmis::Form::Definition::DATA_COLLECTION_FEATURE_ROLES.map do |role|
+      instance_scope = Hmis::Form::Instance.with_role(role).active.published
+      # Service instances must specify a service type or category.
+      instance_scope = instance_scope.for_services if role == :SERVICE
+
+      # Choose the "best" instance, i.e. the one that would actually be selected when recording new data.
+      # We need to do this so that we can accurately set "data collected about" based on the most applicable form.
+      best_instance = instance_scope.detect_best_instance_for_project(project: project)
+
+      has_any_data = case role
+      when :CURRENT_LIVING_SITUATION
+        current_living_situations.exists?
+      when :SERVICE
+        services.exists?
+      when :CE_EVENT
+        events.exists?
+      when :CE_ASSESSMENT
+        assessments.exists?
+      when :CASE_NOTE
+        custom_case_notes.exists?
+      when :REFERRAL
+        false # Referrals are resolved only on projects, not enrollments
+      when :REFERRAL_REQUEST
+        false # Referrals are resolved only on projects, not enrollments
+      when :EXTERNAL_FORM
+        false # Not resolved on enrollments
+      else
+        raise "Unexpected data collection feature role: #{role}"
+      end
+
+      next unless best_instance || has_any_data
+
+      # If the best instance doesn't match this enrollment, return, UNLESS there is existing data.
+      # (Example: data is collected for HoH. If HoH changes, you still see past data from the original HoH's enrollment)
+      next if !has_any_data && best_instance.present? && !best_instance.project_and_enrollment_match(project: project, enrollment: self)
+
+      OpenStruct.new(
+        role: role.to_s,
+        id: [id, role, best_instance&.id].join(':'), # Unique ID for Apollo caching
+        legacy: has_any_data && !best_instance,
+        data_collected_about: best_instance&.data_collected_about || 'ALL_CLIENTS', # Doesn't really matter for legacy
```

Comment:
Maybe it doesn't make sense to return this field at all for enrollment features?

Reply:
Maybe- let's review how we're using this in the frontend now (I think, not at all) - and whether we want the frontend to use it in at ideal world.

Reply:
Sounds good. I checked the [frontend PR](https://github.com/greenriver/hmis-frontend/pull/972/files) to review every usage. (That PR replaces usages of `enrollment.project.dataCollectionFeatures` with the new `enrollment.dataCollectionFeatures` so I think it should be comprehensive)
- EnrollmentRoute access wrapper - `dataCollectedAbout` is not used
- EnrollmentCaseNotes, EnrollmentQuickActions, EnrollmentCeAssessmentsPage, EnrollmentCeEventsPage, EnrollmentCurrentLivingSituationsPage - all of these don't use `dataCollectedAbout`, they only check if a feature is `legacy`. (Maybe a new custom hook is in order?)
- EnrollmentDashboard - `dataCollectedAbout` is not used

Do we want the frontend to use it in an ideal world?
- Perhaps not, bc a data collection feature should only be returned on an enrollment at all if "data collected about" is true for this enrollment. At the frontend side, we don't need to ask "who should this data be collected about in general?" when we already have the answer to "is this data collected about this individual?"
- Otoh, would we ever want to add some instructional text that helps users understand the customizability, like understanding why a tab/form appears when it does? Something like "Collect case notes here for user XX because this project is type YY and the user is the HoH." I have a hard time seeing us adding this because I think the target audience of this instructional text probably wouldn't be the same person who has permission to configure data collection?
- Is there something else I'm missing around complex situations where multiple instances apply to an enrollment? My instinct is still saying that all that complexity should live on the backend only.

Reply:
But one nice thing about leaving it as-is is that we can use the same type `DataCollectionFeatureFieldsFragment` for both project and enrollment

Reply:
Hm, by the same logic, should we exclude `CE_EVENT`, `CE_ASSESSMENT` and `CASE_NOTE` from being resolved on Project? https://github.com/greenriver/hmis-warehouse/blob/martha/6159-data-collection/drivers/hmis/app/models/hmis/hud/project.rb#L242-L247 (and maybe referral and referral request as well, though we already talked about these and decided to include them?)

Reply:
DataCollectionFeatures.dataCollectedAbout used here:
https://github.com/greenriver/hmis-frontend/blob/ccf0f5de4a293ca3f301eaba78290e981a8fd92e/src/modules/enrollment/components/pages/EnrollmentDashboard.tsx#L46-L60
(which calls [this util fn](https://github.com/greenriver/hmis-frontend/blob/ccf0f5de4a293ca3f301eaba78290e981a8fd92e/src/modules/hmis/hmisUtil.ts#L567-L588) to evaluate DataCollectedAbout against the Enrollment).

So based on that, my understanding is that DataCollectedAbout from the project feature object IS currently taken into account when determining whether to show a given `navItem`. (That happens [here](https://github.com/greenriver/hmis-frontend/blob/ccf0f5de4a293ca3f301eaba78290e981a8fd92e/src/modules/enrollment/hooks/useEnrollmentDashboardNavItems.tsx#L44-L46)). In other words, if the `project.dataCollectionFeatures` includes `CURRENT_LIVING_SITUATION` with dca `HOH`, then the CLS tab won't show up on a non-HoH Enrollment dashboard. So it's working as expected (the old behavior, I mean).

I agree that the _new_ approach is to evaluate the "data collected about" logic on the backend, since we are now resolving data collection features on the Enrollment itself. As a result, we choose to just exclude the feature if it's not enabled for this context. So we should be able to pull out most of that logic from the frontend.

I agree we should still resolve `data_collected_about` on the Enrollment feature, for future explanatory text and to use the same type for both.

Reply:
Ok, yes I agree with your understanding of both the old behavior and the new approach. I'm good with the decisions to keep resolving `data_collected_about` here and keep resolving all features on the Project. 👍